### DISCUSSION TITLE
fastlane deliver: "The API key in use does not allow this request"

### DISCUSSION THREAD

**Discussion by yolpsoftware** [fastlane/fastlane#18391] [Q&A]
I'm trying to use `fastlane deliver` to upload an IPA build to an existing app in App Store Connect. This is my command:

```
fastlane deliver --ipa "./my.ipa" --skip_screenshots --skip_metadata
```

This is the error that appears after a while:

```
[!] The request could not be completed because:
	This request is forbidden for security reasons - The API key in use does not allow this request
```

The command runs on an account with Developer Role.

Any idea what could be wrong? I thought that the Developer Role should be sufficient when using `skip_screenshots` and `skip_metadata`.

**Comment by dinsen** [comment]
Have you tried a key with App Manager role to see if that works?

**Reply by simondelphia** [reply]
How do I swap out the API key? Not even sure where it's getting it from. (I'm trying to use Spaceship at the moment which says api keys aren't even supported yet… and I made sure the user I'm using has the "App Manager" role)

**Reply by dinsen** [reply]
You need a user with Admin role to create the API key here: https://appstoreconnect.apple.com/access/api
The Account Holder needs to Request access first if it's not done yet.

You will find a lot more information about the API key here: https://docs.fastlane.tools/app-store-connect-api/

**Reply by simondelphia** [reply]
I am able to create an api key without issue, but I don't understand why spaceship is even giving this error and how to swap in a new api key if that's even the problem.

**Reply by simondelphia** [reply]
I made [another discussion post](https://github.com/fastlane/fastlane/discussions/19427) with some more detail

**Comment by jeremie-l** [comment]
I just noticed that an API key with a Developer Role is not allowed to create a new version of the app.
So, you have to create the  new version in App Store Connect first, and then you can use fastlane to upload a build for this version.
Otherwise, you get this error.

**Reply by finorix079** [reply]
Does that mean if you have a higher-level role (e.g. App Manager) then you can just upload without manually creating a new version?

**Reply by shaverdaTH** [reply]
Yes, @terryjiang2020, it does.
